dotnetv4/S3/Scenarios/S3_CreatePresignedPost/CreatePresignedPostBasics.cs

```diff
@@ -0,0 +1,326 @@
+/// Scenario demonstrating the complete workflow for presigned POST URLs:
+/// 1. Create an S3 bucket
+/// 2. Create a presigned POST URL
+/// 3. Upload a file using the presigned POST URL
+/// 4. Clean up resources
+/// </summary>
+public class CreatePresignedPostBasics
+{
+    public static ILogger<CreatePresignedPostBasics> _logger = null!;
+    public static S3Wrapper _s3Wrapper = null!;
+    public static UiMethods _uiMethods = null!;
+    public static bool _isInteractive = true;
+    public static string? _bucketName;
+    public static string? _objectKey;
+
+    /// <summary>
+    /// Set up the services and logging.
+    /// </summary>
+    /// <param name="host">The IHost instance.</param>
+    public static void SetUpServices(IHost host)
+    {
+        var loggerFactory = LoggerFactory.Create(builder =>
+        {
+            builder.AddConsole();
+        });
+        _logger = new Logger<CreatePresignedPostBasics>(loggerFactory);
+
+        // Now the client is available for injection.
+        _s3Wrapper = host.Services.GetRequiredService<S3Wrapper>();
+        _uiMethods = new UiMethods();
+    }
+
+    /// <summary>
+    /// Perform the actions defined for the Amazon S3 Presigned POST scenario.
+    /// </summary>
+    /// <param name="args">Command line arguments.</param>
+    /// <returns>A Task object.</returns>
+    public static async Task Main(string[] args)
+    {
+        // Check for non-interactive mode
+        _isInteractive = !args.Contains("--non-interactive");
+        
+        // Set up dependency injection for Amazon S3
+        using var host = Microsoft.Extensions.Hosting.Host.CreateDefaultBuilder(args)
+            .ConfigureServices((_, services) =>
+                services.AddAWSService<IAmazonS3>()
+                    .AddTransient<S3Wrapper>()
+            )
+            .Build();
+
+        SetUpServices(host);
+        
+        try
+        {
+            // Display overview
+            _uiMethods.DisplayOverview();
+            _uiMethods.PressEnter(_isInteractive);
+            
+            // Step 1: Create bucket
+            await CreateBucketAsync();
+            _uiMethods.PressEnter(_isInteractive);
+            
+            // Step 2: Create presigned URL
+            _uiMethods.DisplayTitle("Step 2: Create presigned POST URL");
+            var response = await CreatePresignedPostAsync();
+            _uiMethods.PressEnter(_isInteractive);
+            
+            // Step 3: Display URL and fields
+            _uiMethods.DisplayTitle("Step 3: Presigned POST URL details");
+            CreatePresignedPost.DisplayPresignedPostFields(response);
+            _uiMethods.PressEnter(_isInteractive);
+            
+            // Step 4: Upload file
+            _uiMethods.DisplayTitle("Step 4: Upload test file using presigned POST URL");
+            await UploadFileAsync(response);
+            _uiMethods.PressEnter(_isInteractive);
+            
+            // Step 5: Verify file exists
```

Comment:
verifying it exists isnt technically necessary but i added this to show that the file was uploaded